extension/jaegerremotesampling/factory.go

```diff
@@ -5,18 +5,43 @@ package jaegerremotesampling // import "github.com/open-telemetry/opentelemetry-
 
 import (
 	"context"
+	"sync"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configgrpc"
 	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/featuregate"
+	"go.uber.org/zap"
 
+	deprecated "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/deprecated"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal/metadata"
 )
 
+var once sync.Once
+
+func logDeprecation(logger *zap.Logger) {
+	once.Do(func() {
+		logger.Warn("jaegerremotesampling extension will depricate Thrift-gen and replace it with Proto-gen to be compatbible to jaeger 1.42.0 and higher. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18485 for more details.")
+	})
+}
+
+const protoInsteadOfThrift = "extension.jaegerremotesampling.replaceThriftWithProto"
```

Comment:
Neat. Is this something that one can just specify in their extension YAML config? I'm not up to speed on how these work.